libs/domain/cart/cart-remove/cart-remove.component.ts

```diff
@@ -0,0 +1,92 @@
+import { CartService } from '@spryker-oryx/cart';
+import { resolve } from '@spryker-oryx/di';
+import { NotificationService } from '@spryker-oryx/site';
+import { AlertType } from '@spryker-oryx/ui';
+import { ButtonColor, ButtonSize, ButtonType } from '@spryker-oryx/ui/button';
+import { IconTypes } from '@spryker-oryx/ui/icon';
+import { I18nMixin, hydrate } from '@spryker-oryx/utilities';
+import { LitElement, TemplateResult, html } from 'lit';
+import { state } from 'lit/decorators.js';
+import { CartComponentMixin } from '../src/mixins';
+
+@hydrate({ event: ['mouseover', 'focusin'] })
+export class CartRemoveComponent extends CartComponentMixin(
+  I18nMixin(LitElement)
+) {
+  protected cartService = resolve(CartService);
+  protected notificationService = resolve(NotificationService);
+
+  @state() protected requestsConfirmation = false;
+  @state() protected isLoading = false;
+
+  protected override render(): TemplateResult | void {
+    if (!this.$cart()?.id) return;
+
+    return html`
+      <oryx-button
+        .type=${ButtonType.Icon}
+        .size=${ButtonSize.Md}
+        .label=${this.i18n('carts.remove')}
+        .icon=${IconTypes.Trash}
+        @click=${this.onRemove}
+      ></oryx-button>
+      ${this.renderRemoveConfirmationModal()}
+    `;
+  }
+
+  protected onRemove(): void {
+    this.requestsConfirmation = true;
+  }
```

Comment:
```suggestion
   protected onRemove(e: PointerEvent): void {
    if (e.altKey) {
      this.onConfirm();
    } else {
      this.requestsConfirmation = true;
    }
  }
```

What do you think about this? It also might inspire to rename `requestsConfirmation` to just `remove`

Reply:
Nice idea with instant remove, but not sure about renaming of state prop, in general flow we are **requesting confirmation** (ie show the modal), it is not yet exact remove